### PR TITLE
Precompilation does not work for applications using relative path

### LINKED ISSUE #27: Precompilation does not work for applications using relative path
I have structured my ASP.NET Core project according to "Feature Slices" as outlined here:

[https://msdn.microsoft.com/en-us/magazine/mt763233.aspx](https://msdn.microsoft.com/en-us/magazine/mt763233.aspx)

The result is that the views are stored alongside the relevant models and controllers (and not in the "Views" folder). The _Layout file is stored at "Features/Shared/_Layout.cshtml".

When I run the site from Visual Studio or publish the site without razor precompilation enabled, the site works fine. When I enable precompilation I get an error in the console:

```
The layout view 'Shared/_Layout.cshtml' could not be located. The following locations were searched:
Shared/_Layout.cshtml
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.GetLayoutPage(ViewContext context, String executingFilePath, String layoutPath)

```
I have changed the publish options to output the relevant cshtml files so they definitely are included in the publish destination even though I know this technically isn't required when using precompilation (this is also verified since the site works fine without the precompilation enabled).

Is having an alternate location for the Views supported in the precompilation scenario?

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #27
Adds tests for #33

### CODE REVIEW

**Comment by pranavkm** [comment]
Redoing https://github.com/aspnet/MvcPrecompilation/pull/32 after it's addressed PR comments + rebased on dev.

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/PublishWithEmbedViewSourcesTest.cs [hunk 1]
```diff
@@ -32,9 +32,9 @@ public PublishWithEmbedViewSourcesTest(PublishWithEmbedViewSourcesTestFixture fi
             // Arrange
             var expectedViews = new[]
             {
-                "Areas/TestArea/Views/Home/Index.cshtml",
-                "Views/Home/About.cshtml",
-                "Views/Home/Index.cshtml",
+                "/Areas/TestArea/Views/Home/Index.cshtml",
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/PublishWithEmbedViewSourcesTest.cs]: Why did this have to change?

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/PublishWithEmbedViewSourcesTest.cs]: The absence of the first slash makes the view engine treat this is as a relative path (relative to the current view \ app root when there is no view), as opposed to an app-relative path which is what this is (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs#L530-L534).

Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/PublishWithEmbedViewSourcesTest.cs]: 👍  :shipit:

**Pull request closed without merging**